lib/Widgets/SettingsPage/AbstractSettingsPage.vala

```diff
@@ -21,15 +21,59 @@
  * AbstractSettingsPage is a #Gtk.ScrolledWindow subclass with properties used
  * by other Granite settings widgets.
  */
-
 public abstract class Granite.SettingsPage : Gtk.ScrolledWindow {
+    protected string _icon_name;
+    protected string _title;
+
+    public enum StatusType {
+        ERROR,
+        OFFLINE,
+        SUCCESS,
+        WARNING,
+        NONE
+    }
+
+    /**
+     * Selects a colored icon to be displayed in a Granite.SettingsSidebar
+     */
+    public StatusType status_type { get; set; default = StatusType.NONE; }
+
+    /**
+     * A widget to display in place of an icon in a Granite.SettingsSidebar
+     */
+    public Gtk.Widget? display_widget { get; construct; }
+
+    /**
+     * A header to be sorted under in a Granite.SettingsSidebar
+     */
+    public string? header { get; construct; }
+
+    /**
+     * An status string associated with #this
+     */
+    public string status { get; set; }
+
     /**
      * An icon name associated with #this
      */
-    public string? icon_name { get; construct set; }
+    public string? icon_name {
```

Comment:
Yes, it's because we need the backing variables to be `protected` instead of `private`. This way `SimpleSettingsPage` can access these variables and the cast in `SettingsSidebar` works as expected